src/Agent.Worker/JobRunner.cs

```diff
@@ -218,6 +218,32 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     }
                 }
 
+                // Expand sidecar container properties
+                foreach (var sidecar in jobContext.SidecarContainers)
```

Comment:
maybe we can add a method to ContainerInfo and let it expand variables.
```
void ExpandVariables(jobContext.Variables)
{
}
```
any reason we don't do this for job container?

Reply:
Functionality is there if we want to add ports and user provided volumes to JobContainer later, but out of scope of this feature